tools/src/main/java/org/apache/kafka/tools/ShareConsumerPerformance.java

```diff
@@ -0,0 +1,465 @@
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+            shareConsumersConsumptionDetails.add(shareConsumerConsumption);
+        }
+        LOG.debug("Shutting down of thread pool is started");
+        executorService.shutdown();
+
+        try {
+            // Wait a while for existing tasks to terminate.
+            // Adding 100 ms to the timeout so all the threads can finish before we reach this part of code.
+            if (!executorService.awaitTermination(recordFetchTimeoutMs + 100, TimeUnit.MILLISECONDS)) {
+                LOG.debug("Shutting down of thread pool could not be completed. It will retry cancelling the tasks using shutdownNow.");
+                executorService.shutdownNow(); // Cancel currently executing tasks
+                // Wait a while for tasks to respond to being cancelled
+                if (!executorService.awaitTermination(recordFetchTimeoutMs + 100, TimeUnit.MILLISECONDS))
+                    LOG.debug("Shutting down of thread pool could not be completed even after retrying cancellation of the tasks using shutdownNow.");
+            }
+        } catch (InterruptedException e) {
+            // (Re-)Cancel if current thread also interrupted
+            LOG.warn("Encountered InterruptedException while shutting down thread pool. It will retry cancelling the tasks using shutdownNow.");
+            executorService.shutdownNow();
+            // Preserve interrupt status
+            Thread.currentThread().interrupt();
+        }
+
+        if (options.showShareConsumerStats()) {
+            long endMs = System.currentTimeMillis();
+            for (int index = 0; index < shareConsumersConsumptionDetails.size(); index++) {
+                // Print stats for share consumer.
+                double elapsedSec = (endMs - startMs) / 1_000.0;
+                long fetchTimeInMs = endMs - startMs;
+                long messagesReadByConsumer = shareConsumersConsumptionDetails.get(index).messagesConsumed();
+                long bytesReadByConsumer = shareConsumersConsumptionDetails.get(index).bytesConsumed();
+                printStats(bytesReadByConsumer, messagesReadByConsumer, elapsedSec, fetchTimeInMs, startMs, endMs, options.dateFormat(), index + 1);
+            }
+        }
+
+        if (messagesRead.get() < numMessages) {
+            System.out.printf("WARNING: Exiting before consuming the expected number of messages: timeout (%d ms) exceeded. " +
+                    "You can use the --timeout option to increase the timeout.%n", recordFetchTimeoutMs);
+        }
+        totalMessagesRead.set(messagesRead.get());
+        totalBytesRead.set(bytesRead.get());
+    }
+
+    private static void consumeMessagesForSingleShareConsumer(long currentTimeMs,
```

Comment:
This could be more neatly initialised by having `lastConsumedTimeMs` and `currentTimeMs` as local variables, and then initialising with
```
long currentTimeMs = System.currentTimeMills();
long lastConsumedTimeMs = currentTimeMs;
```
and then going into the while loop.